cpp/src/arrow/util/thread_pool.cc

```diff
@@ -321,13 +321,28 @@ void ThreadPool::CollectFinishedWorkersUnlocked() {
   state_->finished_workers_.clear();
 }
 
+thread_local ThreadPool* g_current_thread_pool = nullptr;
+
+ThreadPool* ThreadPool::GetCurrentThreadPool() { return g_current_thread_pool; }
+
+thread_local int g_current_thread_id = 0;
+
+int ThreadPool::GetCurrentThreadId() { return g_current_thread_id; }
```

Comment:
If this is returning the thread's `index` maybe `id` is not the best name?  Especially since it might be confused with the existing concept of a thread id.

Reply:
Good call, I'll rename to thread index